File: ruby/find_numbers_with_even_number_of_digits.rb
# https://leetcode.com/problems/find-numbers-with-even-number-of-digits/

def find_numbers(nums)
  even_digit_count = 0
  nums.each do |num|
    even_digit_count += 1 if num.digits.count.even?
  end

  even_digit_count
end

File: python/find_numbers_with_even_number_of_digits.py
# https://leetcode.com/problems/find-numbers-with-even-number-of-digits/

class Solution:
    def findNumbers(self, nums):
        even_numbers = 0
        for num in nums:
            if len(str(num)) % 2 == 0:
                even_numbers += 1

        return even_numbers


result = Solution.findNumbers(Solution, [12,345,2,6,7896])
print(result)